AxonDeepSeg/segment.py

```diff
@@ -127,6 +127,19 @@ def segment_image(
     path_testing_image = convert_path(path_testing_image)
     path_model = convert_path(path_model)
 
+    # Get pixel size from file, if needed
+    # If we did not receive any resolution we read the pixel size in micrometer from each pixel.
+    if acquired_resolution == None:
+        if (Path(path_testing_image.parents[0]) / 'pixel_size_in_micrometer.txt').exists():
+            resolutions_file = open(Path(path_testing_image.parents[0]) / 'pixel_size_in_micrometer.txt', 'r')
+            str_resolution = float(resolutions_file.read())
+            acquired_resolution = float(str_resolution)
+        else:
+            exception_msg = "ERROR: No pixel size is provided, and there is no pixel_size_in_micrometer.txt file in image folder. " \
+                            "Please provide a pixel size (using argument acquired_resolution), or add a pixel_size_in_micrometer.txt file " \
+                            "containing the pixel size value."
+            raise Exception(exception_msg)
```

Comment:
Shouldn't we use `sys.exit(3)` again for this? It's essentially the same error.

Reply:
I'll check to see if doing that makes any test fail; I don't recall why we wrote it this way here. Maybe because here we're not calling directly through a command line (i.e. to reach this error, we *know* it will be a Python call to the function, so raising a Python exception allows us to catch and throw it in other loops, while in the other case we wrote the error inside the CLI call, so we'll always want to exit it completely).

Reply:
But, I wouldn't dwell too much on these tangential details, we'll start to go off track from the aim of this PR (which is the zoom factor); new issues could be opened to deal with this type of question/inconsistency.

Reply:
I agree (the PR is already bigger than expected)

For what it's worth this error was probably never thrown (except in the edge case of calling directly `segment_image`) because the pixel size text files are already read inside the main of segment.py (before the call to `segment_image`/`segment_folder`). So, in total, we read the text files in 4 different places in segment.py.